Gruntfile.js

```diff
@@ -286,6 +289,11 @@ module.exports = function(grunt) {
     nodeunit: {
       all: ['tests/nodeunit/unit/**/*.js']
     },
+    mochaTest: {
+      integration: {
+        src: ['tests/integration/**/*.js'],
+      },
+    },
```

Comment:
Karma, Mocha, Nodeunit and Protractor. I wonder if we can merge some of these. Not for this commit necessarily, but perhaps Nodeunit can be replaced with Mocha? Not completely sure the differences between each. 

As In understand it:
 - Karma has frameworks around you testing code inside Angular etc, and so should be used for code that you expect to run in the browser
 - Protractor is how you click things, and so is for end to end tests
 - Nodeunit is for random JS testing that nicely supports callbacks
 - Mocha is… a better nodeunit? I can't think of what Nodeunit does that Mocha doesn't.